doc/src/MIOpen_Porting_Guide.md

```diff
@@ -0,0 +1,2041 @@
+
+# MIOpen Porting Guide
+
+
+## The key differences between MIOpen and cuDNN:
+* MIOpen only supports 4-D tensors in the NCHW storage format. This means all the __“\*Nd\*”__ APIs in cuDNN do not have a corresponding API in MIOpen.
+* MIOpen only supports __`float(fp32)`__ data-type.
+* MIOpen only supports __2D Convolutions__ and __2D Pooling__.
+* Calling miopenFindConvolution*Algorithm() is *mandatory* before calling any Convolution API.
+* Typical calling sequence for Convolution APIs for MIOpen is:
+    * miopenConvolution*GetWorkSpaceSize() // returns the workspace size required by Find()
+    * miopenFindConvolution*Algorithm() // returns performance info about various algorithms
+    * miopenConvolution*()
+* MIOpen does not support __Preferences__ for convolutions.
+* MIOpen does not support Softmax modes. MIOpen implements the __SOFTMAX_MODE_CHANNEL__ flavor.
+* MIOpen does not support __Transform-Tensor__, __Dropout__, __RNNs__, and __Divisive Normalization__.
+
+<br/><br/><br/><br/>
+
+## Helpful MIOpen Environment Variables
+`MIOPEN_ENABLE_LOGGING=1` – log all the MIOpen APIs called including the parameters passed to
+those APIs. \
+`MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES=0` – disable Winograd convolution
+algorithm. \
+`MIOPEN_DEBUG_GCN_ASM_KERNELS=0` – disable hand-tuned asm. kernels for Direct convolution
+algorithm. Fall-back to kernels written in high-level language. \
+`MIOPEN_DEBUG_CONV_FFT=0` – disable FFT convolution algorithm. \
+`MIOPEN_DEBUG_CONV_DIRECT=0` – disable Direct convolution algorithm.
```

Comment:
The _Helpful MIOpen Environment Variables_ section from doc/src/MIOpen_Porting_Guide.md must be revised.

- duplicates info from [DebugAndLogging.md](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/doc/src
- outdated (e.g. description of MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES is incorrect)
- incomplete

Reply:
@atamazov I will create a separate PR for this.